### DISCUSSION TITLE
ONNX Gather not a graph node in ezkl?

### DISCUSSION THREAD

**Discussion by biancaganescu** [zkonduit/ezkl#236] [Q&A] [answered]
I've been trying to use nn.Embedding with ezkl, but it does not work and I am not sure where to point the problem. Looking into it, ONNX maps nn.Embedding to the Gather operation (which I am not sure it gives the desired output anyways in that case) - from [torch ONNX repo](https://github.com/pytorch/pytorch/blob/ef5e2481ce353255d5b23d6930426c6486aa8721/torch/onnx/symbolic_opset9.py#L482). Then ezkl does not recognise "Gather" as node 0 in the graph. [sonos/tract](https://github.com/sonos/tract) say they implement "Gather", but I assume that is not a node for them so therefore that is not a node for ezkl?

This is the ONNX graph created for this code, with input x = torch.ones(1, 64, dtype=int):
```
class Model(nn.Module):
    def __init__(self):
        super().__init__()
        self.layer = nn. Embedding(65, 64)

    def forward(self, x):
        return self.layer(x)

```

<img width="315" alt="Screenshot 2023-05-12 at 19 42 35" src="https://github.com/zkonduit/ezkl/assets/59770285/d252eab8-c7cd-4b93-b4dd-8031fcd917df">

And this is the ezkl error for it:
<img width="1013" alt="Screenshot 2023-05-12 at 19 44 20" src="https://github.com/zkonduit/ezkl/assets/59770285/ccf4d257-a3bb-41da-bd8c-a9ef40f877ce">

**Comment by alexander-camuto** [comment]
thanks for this let me take a look :)

**Reply by alexander-camuto** [reply, marked as answer]
As discussed offline -- this has been resolved with latest updates.
